### PR TITLE
SMS based opt-out 

### PR DESCRIPTION
**Pull request by imsickofmaps**

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@hodgestar ready for review

File: test/smsinbound.test.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                    .setup.user.addr('+27001')
+                    .setup.user.state('states_start')
+                    .input('STOP')
+                    .check.interaction({
+                        state: 'states_opt_out',
+                        reply: 
+                            'Thank you. You will no longer receive messages from us. ' +
+                            'If you have any medical concerns please visit your nearest clinic'
+                    })
+                    .run();
```
Comment (hodgestar) [inline test/smsinbound.test.js:129]: I think we also need to check that the optout was performed? I.e. that the API calls in the on:enter function happened?

File: test/smsinbound.test.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                return tester
+                    .setup.user.addr('+27001')
+                    .setup.user.state('states_start')
+                    .input('START')
+                    .check.interaction({
+                        state: 'states_opt_in',
+                        reply: 
+                            'Thank you. You will now receive messages from us again. ' +
+                            'If you have any medical concerns please visit your nearest clinic'
+                    })
```
Comment (hodgestar) [inline test/smsinbound.test.js:144]: Same comment here, except we should check that the opt-in happened.

**Comment by hodgestar** [comment]
Otherwise looks good.

**Comment by imsickofmaps** [comment]
So in testing this on the live line to see if it worked as expected, it looks like Vumi Go hijacks the STOP SMS and I don't even see it in my conversation/app.

**Pull request merged**